package-scripts/datadog-signing-keys/postinst

```diff
@@ -23,14 +25,29 @@ USE_TRUSTED_GPG_D=true
 DISTRIBUTION_VERSION=$(. /etc/os-release && echo "${VERSION_ID}" | tr "." " " | awk '{ print $1 }')
 
 if { [ "${DISTRIBUTION}" = "Debian" ] && [ "${DISTRIBUTION_VERSION}" -ge "9" ]; } || { [ "${DISTRIBUTION}" = "Ubuntu" ] && [ "${DISTRIBUTION_VERSION}" -ge "16" ]; } ; then
+    DD_LIST_SIGNED_BY=false
+    OPW_LIST_SIGNED_BY=false
     # We also want to create the /etc/apt/trusted.gpg.d keyring if the system
     # is not set to use the /usr/share/keyrings/ keyring through signed-by.
     # This might create the /etc/apt/trusted.gpg.d keyring unnecessarily on
     # some systems, but it's the only way to be at least somewhat certain
     # that anyone who installed the package is ok.
-    if grep "signed-by=${USR_SHARE_KEYRING}" /etc/apt/sources.list.d/datadog.list 2>/dev/null 1>/dev/null; then
+    if [ ! -e "${DD_SOURCE_LIST}" ] || grep "signed-by=${USR_SHARE_KEYRING}" "${DD_SOURCE_LIST}" 2>/dev/null 1>/dev/null; then
+        DD_LIST_SIGNED_BY=true
+    fi
+
+    if [ ! -e "${OPW_SOURCE_LIST}" ] || grep "signed-by=${USR_SHARE_KEYRING}" "${OPW_SOURCE_LIST}" 2>/dev/null 1>/dev/null; then
+        OPW_LIST_SIGNED_BY=true
+    fi
+
+    if [ "${DD_LIST_SIGNED_BY}" = "true" ] && [ "${OPW_LIST_SIGNED_BY}" = "true" ]; then
         USE_TRUSTED_GPG_D=false
     fi
+
+    # See https://github.com/DataDog/datadog-signing-keys/pull/27#discussion_r1028474944
+    if [ ! -e "${DD_SOURCE_LIST}" ] && [ ! -e "${OPW_SOURCE_LIST}" ]; then
+        USE_TRUSTED_GPG_D=true
+    fi
 fi
```

Comment:
I believe the following change makes it easier to see the "if at least one known source file is detected and contains the signed-by option, then we don't add the keyring in `/etc/apt/trusted.gpg.d`" logic.

```suggestion
    # See https://github.com/DataDog/datadog-signing-keys/pull/27#discussion_r1028474944
    if grep "signed-by=${USR_SHARE_KEYRING}" "${DD_SOURCE_LIST}" 2>/dev/null 1>/dev/null; then
        USE_TRUSTED_GPG_D=false
    fi

    if grep "signed-by=${USR_SHARE_KEYRING}" "${OPW_SOURCE_LIST}" 2>/dev/null 1>/dev/null; then
        USE_TRUSTED_GPG_D=false
    fi
fi
```

Reply:
Right, I misread that comment, sorry for that. Thanks for the thorough review, I'll merge now.